src/modules/UI/scenes/Plugins/EdgeProvider.js

```diff
@@ -349,10 +351,16 @@ export class EdgeProvider extends Bridgeable {
     console.log(`requestSpendUri ${uri}`)
     const { currencyWallets } = this._state.core.account
     const edgeWallet = currencyWallets[this._state.ui.wallets.selectedWalletId]
-    const result = await edgeWallet.parseUri(uri)
 
+    const result: EdgeParsedUri & { paymentProtocolURL?: string } = await edgeWallet.parseUri(uri)
     const { currencyCode = result.currencyCode, customNetworkFee, metadata, lockInputs = true, uniqueIdentifier, orderId } = options
 
+    // Check is PaymentProtocolUri
+    if (!!result.paymentProtocolURL && !result.publicAddress) {
+      await launchBitPay(result.paymentProtocolURL, { wallet: edgeWallet, metadata }).catch(showError)
```

Comment:
Perfect. I will take a look at `launchBitPay` to see if it can be modified to better conform with the full payment protocol spec. Ideally it will eventually be renamed throughout to something like `launchPayProtocol`.